### PR TITLE
csi-manila: deployment manifests refactoring + helm chart

### PR DESCRIPTION
**Pull request by gman0**
**What this PR does / why we need it**:
This PR adds no new features to csi-manila, but only adds a Helm chart under `examples/manila-csi-plugin/helm-deployment` and refactors deployment manifests + updates the docs accordingly.

Manifests in `manifests/manila-csi-plugin` are generated from the Helm chart.

Why is the Helm chart needed:
* as it's mentioned in the [documentation](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-manila-csi-plugin.md#deployment), CSI Manila may serve only a single share protocol per plugin instance. If an operator wants access to multiple share protocols, multiple deployments need to be made. This requires numerous changes to the manifests which can be quite cumbersome
* the chart solves this by deploying the driver + its sidecar containers per share protocol (see `examples/manila-csi-plugin/helm-deployment/values.yaml`)

This PR also addresses comments from https://github.com/kubernetes/cloud-provider-openstack/pull/652 where we agreed to do changes regarding deployment in a separate PR (i.e. this one), notably:
* changes to manifests requested in https://github.com/kubernetes/cloud-provider-openstack/pull/652#discussion_r302387455 , https://github.com/kubernetes/cloud-provider-openstack/pull/652#discussion_r302388003 , https://github.com/kubernetes/cloud-provider-openstack/pull/652#discussion_r302389332 
* docs updated to address https://github.com/kubernetes/cloud-provider-openstack/pull/652#discussion_r302387565

We're now following similar "2-component" deployment as CSI Cinder does:
* Controller plugin (StatefulSet) now contains the CSI Manila plugin itself along with csi-provisioner, csi-snapshotter. csi-attacher has been removed because volume attachments are not needed. CSIDriver CRD is used to pass the `attachRequired: false` option to k8s,
* Node plugin (DaemonSet) contains the CSI Manila plugin and csi-node-driver-registrar

```release-note
CSI Manila: added a Helm chart
```

### CODE REVIEW

**Comment by gman0** [comment]
PTAL @tsmetana @adisky @ramineni

**Comment by hogepodge** [comment]
I'm fully in support of this.

File: docs/using-manila-csi-plugin.md [hunk 1]
```diff
@@ -61,35 +61,46 @@ Optionally, a custom certificate may be sourced via `os-certAuthorityPath` (path
...
+
+The deployment consists of two main components: Controller and Node plugins along with their respective RBACs. Controller plugin is deployed as a StatefulSet and runs CSI Manila, [external-provisioner](https://github.com/kubernetes-csi/external-provisioner) and [external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter). Node plugin is deployed as a DaemonSet and runs CSI Manila and [csi-node-driver-registrar](https://github.com/kubernetes-csi/node-driver-registrar).
+
+**Deploying with Helm**
+
+This is the preferred way of deployment because it greatly simplifies the difficulties with managing multiple share protocols.
+
+CSI Manila Helm chart is located in `examples/manila-csi-plugin/helm-deployment`.
+
+First, modify `values.yaml` to suite your environment, and then simply install the Helm chart with `$ helm install helm-deployment`.
```
Comment (gouthampacha) [inline docs/using-manila-csi-plugin.md]: We could add a note here about naming, due to the way the resources in our manifests are named, the helm release name should be 12 characters or less? Alternatively, we could find a way to shorten the resource names by dropping the "plugin" in the names? just a thought..

**Comment by tombarron** [comment]
Nice work, and thanks for addressing gouthampacha's comments.  The helm chart makes it a lot more straightforward to do multi-protocol deployments!

**Comment by lingxiankong** [comment]
/lgtm

thanks for the work, it's really helpful for the users.

**Pull request merged**